src/unification.hpp

```diff
@@ -0,0 +1,36 @@
+#pragma once
+
+#include <vector>
+#include <functional>
+
+namespace Unification {
+
+struct Core {
+
+	enum class Tag { Var, Term, };
+
+	struct NodeHeader {
+		Tag tag;
+		int data_idx;
+	};
+
+	struct TermData {
+		int function_id; // possibly an external id
+		std::vector<int> argument_idx;
+	};
+
+	std::function<void(int,int)> unify_function;
```

Comment:
This doesn't take function IDs, but node IDs, right?

Reply:
Just trying to understand what you changed

Reply:
Yes, it now takes "header references"